tests/saw/spec/handshake/handshake_io_lowlevel.saw

```diff
@@ -187,14 +212,22 @@ let setup_connection = do {
                             ((ocsp_flag == 1) && (status_size > 0)) ||
                             ((mode == 1) && (ocsp_flag == 1))
                      ,resume_from_cache = False
-                     ,client_auth_flag = if mode == S2N_CLIENT then client_cert_auth_type == 1 else 
-                                         if mode == S2N_SERVER then client_cert_auth_type != 0 else False
+                     ,client_auth_flag = if mode == S2N_CLIENT then client_cert_auth_type == S2N_CERT_AUTH_REQUIRED else
+                                         if mode == S2N_SERVER then client_cert_auth_type != S2N_CERT_AUTH_NONE     else False
                      ,no_client_cert = no_client_cert != zero
                      ,actual_protocol_version = version
+                     ,early_data_state = early_data_state
+                     ,chosen_psk_null = chosen_psk_null_bit
+                     ,quic_enabled = quic_enabled_bit
                      }
                   }}); 
 };
 
+//Set conn->psk_params.chosen_psk to NULL, which enables the use of FULL_HANDSHAKE.
+let setup_connection_chosen_psk_null = setup_connection true;
+
+//Set conn->psk_params.chosen_psk to a non-NULL value, which disables the use of FULL_HANDSHAKE.
+let setup_connection_chosen_psk_non_null = setup_connection false;
```

Comment:
Maybe use the names "setup_psk_connection" and "setup_connection" instead?

Also, the comments aren't quite right. "chosen_psk" is only relevant to TLS1.3 connections, so pre-TLS1.2 connections can have a null chosen_psk and still not be FULL_HANDSHAKE. TLS1.2 handles session resumption differently.